xbmc/epg/EpgInfoTag.cpp

```diff
@@ -465,6 +489,144 @@ std::string CEpgInfoTag::PlotOutline(bool bOverrideParental /* = false */) const
   return retVal;
 }
 
+void CEpgInfoTag::SetIMDBNumber(const std::string &strIMDBNumber)
+{
+  bool bUpdate(false);
+  {
+    CSingleLock lock(m_critSection);
+    bUpdate = m_strIMDBNumber != strIMDBNumber;
+    m_bChanged |= bUpdate;
+    m_strIMDBNumber = strIMDBNumber;
+  }
+  if (bUpdate)
+    UpdatePath();
+}
+
+std::string CEpgInfoTag::IMDBNumber(bool bOverrideParental /* = false */) const
+{
+  std::string retVal;
+  CSingleLock lock(m_critSection);
+  if (bOverrideParental || !m_pvrChannel || !g_PVRManager.IsParentalLocked(*m_pvrChannel))
```

Comment:
I assume the parental lock check is needed because an add-on could automatically grab pics from the web based on the imdb ID?